src/datasources/TimeseriesDatasource.ts

```diff
@@ -69,35 +68,39 @@ async function findTsByAssetAndRelationships(
   //  we only get the first 100 timeseries, and show a warning if there are too many timeseries
   const limit = 101;
   let timeseriesFromRelationships = [];
-  const timeseriesFromAssets =
-    assetQuery?.includeSubTimeseries !== false
-      ? await getTimeseries({ filter, limit }, connector)
-      : [];
-  if (assetQuery?.withRelationships) {
-    const relationshipsList = await fetchRelationships(
-      assetQuery.relationshipsQuery,
-      timeFrame,
-      limit,
-      connector
-    );
-    timeseriesFromRelationships = relationshipsList.map(({ target }) => target);
-  }
-  if (timeseriesFromAssets.length >= limit) {
-    emitEvent(responseWarningEvent, { refId, warning: TIMESERIES_LIMIT_WARNING });
-    timeseriesFromAssets.splice(-1);
-  }
-  if (timeseriesFromRelationships.length >= limit) {
-    emitEvent(responseWarningEvent, { refId, warning: TIMESERIES_LIMIT_WARNING });
-    timeseriesFromRelationships.splice(-1);
+  try {
+    const timeseriesFromAssets =
+      assetQuery?.includeSubTimeseries !== false
+        ? await getTimeseries({ filter, limit }, connector)
+        : [];
+    if (assetQuery?.withRelationships) {
+      const relationshipsList = await fetchRelationships(
+        assetQuery?.relationshipsQuery,
+        connector,
+        timeFrame
+      );
+      timeseriesFromRelationships = _.map(relationshipsList, 'target');
+    }
+    if (timeseriesFromAssets.length >= limit) {
+      emitEvent(responseWarningEvent, { refId, warning: TIMESERIES_LIMIT_WARNING });
+      timeseriesFromAssets.splice(-1);
+    }
+    if (timeseriesFromRelationships.length >= limit) {
+      emitEvent(responseWarningEvent, { refId, warning: TIMESERIES_LIMIT_WARNING });
+      timeseriesFromRelationships.splice(-1);
+    }
+    return _.uniqBy([...timeseriesFromAssets, ...timeseriesFromRelationships], 'id');
+  } catch (error) {
+    handleError(error, refId);
+    return [];
```

Comment:
there is some error handling later in the stack, why do we need it here?
this might then call datapoints api with an empty list, which doesn't make sense anyway. so it's better to break the logic here